_posts/2021/2021-03-31-hmat_memkind.md

```diff
@@ -0,0 +1,172 @@
+| **MEMKIND_HIGHEST_CAPACITY_PREFERRED**		    |   Capacity                 |		
+| **MEMKIND_HIGHEST_CAPACITY_LOCAL**            |   Capacity                 |
+| **MEMKIND_HIGHEST_CAPACITY_LOCAL_PREFERRED**  |   Capacity                 |
+| **MEMKIND_LOWEST_LATENCY_LOCAL**              |   Latency                  |
+| **MEMKIND_LOWEST_LATENCY_LOCAL_PREFERRED**    |   Latency                  |
+| **MEMKIND_HIGHEST_BANDWIDTH_LOCAL**           |   Bandwidth                |
+| **MEMKIND_HIGHEST_BANDWIDTH_LOCAL_PREFERRED**	|   Bandwidth                |
+
+The new semantics based on comparison specific memory property between different NUMA Nodes presented on the platform and choosing
+the best attribute.
+
+### What's new 
+
+**Locality**
+
+So far in memkind, we determined the locality as the nearest NUMA based on the shortest NUMA distance metrics. 
+In this strategy, we miss information about Socket and NUMA Node relation. In the case of non-symmetrical memory topology,
+where for example, PMEM is placed only on one socket, and we would like to avoid remote memory access we could use
+one of the memory kinds from the new API to allocate from memory in local NUMA domain.
+
+<figure class="image">
+  <img src="/assets/memkind_single_pmem_medium_view.png" alt="non-symetric-medium-view">
+  <figcaption>Figure 5. Memory architecture overview based on physical medium.</figcaption>
+</figure>
+
+Using `MEMKIND_HIGHEST_CAPACITY_LOCAL` memkind will allocate to highest capacity Node inside socket:
+
+<figure class="image">
+  <img src="/assets/memkind_single_pmem_property_view_alloc.png" alt="non-symetric-property-view">
+  <figcaption>Figure 6. Memory architecture overview based on memory property.</figcaption>
+</figure>
+
+**New API**
+
+Memory property like bandwidth and latency depends on the specific initiator of the memory request. This results in multidimensional analysis based on the different NUMA Node relationships and [performance characteristics][numa-performance]. Based on this fact, with the new API, memkind addresses different scenarios of potential usage in more complicated memory topologies.
+
+**High Bandwidth**
+
+To help using legacy API: `MEMKIND_HBW*` and determine what value of memory bandwidth could be treated as high bandwidth, we introduced
+`MEMKIND_HBW_THRESHOLD` - a threshold, from which NUMA Node is identified as High Bandwidth Memory NUMA Node.
+
+### What’s Coming Next
+
+NUMA Node concept evolves, and it’s much different compared to the past. No longer, it focuses only on local and remote access. The modern approach based on a concept with the initiator of a memory request and target of a memory request. Both initiator and target are characterized by different performance properties. No longer if some entity can make an allocation request means that it has to be CPU. A similar rule applies to memory – if the processing unit requests memory, it doesn’t mean that memory must come from DRAM.
```

Comment:
The idea of NUMA nodes has evolved, and it's must different now compared to the past.

It no longer focuses on

The modern approach is based on

Memory allocation requests can now be made by a diverse type of XPUs. (is this true? can a GPU allocate DRAM ?)

And the same follows for memory - if something is requesting memory, it doesn't have to come from DRAM.